### PR TITLE
Rpi 4.12.y

### PR DESCRIPTION
**Pull request by VT-Github-Raspberrypi**
ARM64: bcm2837: enable sdhost in device tree for the
Raspberry Pi 3

Fix for https://github.com/raspberrypi/linux/issues/2151

Signed-off-by: VT-Github-Raspberrypi <github.raspberrypi@votarytech.com>

### CODE REVIEW

**Review by VT-Github-Raspberrypi — COMMENTED** [review]
It looks good to me.

**Comment by pelwell** [comment]
screenshot
Really?

**Comment by pelwell** [comment]
And when you fix it, please give your commit a proper name - Rpi 4.12.y isn't very descriptive.

**Comment by VT-Github-Raspberrypi** [comment]
Updated commit message as per review comment by pelwell.

**Comment by VT-Github-Raspberrypi** [comment]
Follow these steps for RPI-3 64bit boot up.
https://github.com/raspberrypi/linux/wiki/config.txt

**Review by VT-Github-Raspberrypi — COMMENTED** [review]
It looks good to me.

**Review by VT-Github-Raspberrypi — COMMENTED** [review]
It looks good to me.

File: arch/arm/boot/dts/bcm2835-rpi.dtsi [hunk 1]
```diff
@@ -27,6 +27,13 @@
 			firmware = <&firmware>;
 			#power-domain-cells = <1>;
 		};
+
```
Comment (pelwell) [inline arch/arm/boot/dts/bcm2835-rpi.dtsi]: This is an upstream file - please don't modify it, except by trying to push changes upstream.

Downstream DT has a framebuffer node in bcm2708-rpi.dtsi.

File: arch/arm64/boot/dts/broadcom/bcm2837-rpi-3-b.dts [hunk 2]
```diff
@@ -22,3 +22,14 @@
 &uart1 {
 	status = "okay";
 };
+
```
Comment (pelwell) [inline arch/arm64/boot/dts/broadcom/bcm2837-rpi-3-b.dts]: This is also an upstream file. Have you considered using bcm2710-rpi-3-b.dts instead?

**Review by pelwell — CHANGES_REQUESTED** [review]

**Comment by VT-Github-Raspberrypi** [comment]
Now reverted frambuffer changes.

**Review by VT-Github-Raspberrypi — COMMENTED** [review]
Now seems all fine.

**Comment by pelwell** [comment]
bcm2837-rpi-3-b.dts is also an upstream file. Have you considered using bcm2710-rpi-3-b.dts instead?

**Comment by VT-Github-Raspberrypi** [comment]
Yes. We only used bcm2837-rpi-3-b.dts.

**Comment by pelwell** [comment]
And I'm saying that I won't accept changes to that file. Try using bcm2710-rpi-3-b.dts instead.

**Comment by VT-Github-Raspberrypi** [comment]
Without this changes Pi-3 board doesn't boot up using bcm2837 dts. 
As per your suggestion like ,"use bcm2710 dts" then What is the use for bcm2837 dts ?.

We have been using bcm2837 dts and making changes according to considering SOC(BCM2837)  raspberry pi-3 64bit.

**Comment by pelwell** [comment]
bcm2837-rpi-3-b.dts is the official upstream Linux support for Pi3 (64-bit, in this case). Upstream changes happen slowly, and they don't support overlays etc., but you can get the source from kernel.org.

bcm2710-rpi-3-b.dts is the downstream Pi3 support, although we don't officially support 64-bit ourselves. Downstream changes can be very quick, but there is likely to be more churn than you get with upstream-only builds.

**Comment by VT-Github-Raspberrypi** [comment]
My changes are on upstream file bcm2837 then what is the reason for not accepting my changes ?.

Let me clear that , What should i do for my changes to be get merged on https://github.com/raspberrypi/linux/tree/rpi-4.12.y ?

**Comment by pelwell** [comment]
Changes to upstream files should be submitted as patches via email to the Linux maintainers. There is a guide to the process here: https://www.kernel.org/doc/html/v4.12/process/submitting-patches.html

**Comment by VT-Github-Raspberrypi** [comment]
What should i do for my changes to be get merged on https://github.com/raspberrypi/linux/tree/rpi-4.12.y ? . 
same process ?.

**Comment by pelwell** [comment]
Once it is accepted upstream (which could be to a 4.13 release candidate, but any branch will do) we can cherry-pick it back to rpi-4.12.y.

**Comment by lategoodbye** [comment]
Please look at my comments in #2151 . This change can't be accepted in upstream because it's already there in 4.13.

**Comment by pelwell** [comment]
Closing as the next RPi kernel version will be 4.14.

**Pull request closed without merging**